### PR TITLE
docs(spirekey): add documentation for connecting a dapp to Kadena SpireKey

### PR DESCRIPTION
**Pull request by alber70g**

### CODE REVIEW

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+A GET request needs to be send to Kadena SpireKey
+
+```
+spirekey.kadena.io/login?returnUrl=dapp.com/auth
+```
+
+- host `spirekey.kadena.io`
+- path `/login`
+- querystring parameters
+  - `returnUrl` the url where the user will be redirected to after
```
Comment (EnoF) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:125]: Maybe just a note for the developers that it should include https protocol

**Review by EnoF — APPROVED** [review]

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+The similarities are:
+
+- the **dApp** is the **Client Application**
+- **Kadena SpireKey** works as a delegate for **Authorization Server** and in
+  turn acesses **WebAuthN** to grant access
+- and the **dApp** is hosted by the **Resource Server**
+
+Let's take a look in how the flow would be for a **dApp** and **Kadena
+Spirekey**
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:53]:
```suggestion

```

I'm finding this hard to follow. See my somewhat mangled work-in-progress replacement text.

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+1. The user creates an account on Kadena SpireKey by authenticating with
+   WebAuthN
+2. The user login with Kadena SpireKey from the dApp
+3. The user wants to sign a transaction created by the dApp
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:58]:
```suggestion
## Integrate with Kadena SpireKey

There are three basic integration points between a decentralized application and Kadena SpireKey.
To integrate with Kadena SpireKey, your application needs to enable users to perform the following steps:

 - Register at least one account on Kadena SpireKey by authenticating using a Web Authentication (WebAuthN) method.
- Log in using a Kadena SpireKey account with authentication provided by a Web Authentication (WebAuthN) method.
- Sign transactions created by the application using a Kadena SpireKey account with authentication provided by a Web Authentication (WebAuthN) method.
```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+### Create account in Kadena SpireKey
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:59]:
```suggestion
### Register an account in Kadena SpireKey
```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:60]:
```suggestion
The following diagram illustrates the process for registering an account using Kadena SpireKey.
```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  alt Validation by blockchain
+    k->>k: executing transaction, mining block
+    k->>s: account created `c:my-account`
+  else Validation failed
+    k->>s: transaction invalid
+  end
+```
+
+### Login with Kadena SpireKey
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:80]:
```suggestion
### Log in with Kadena SpireKey

The following diagram illustrates the process for logging in with a Kadena SpireKey account.
 ```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    u->>d: Login with SpireKey
+  create participant s as Kadena SpireKey
+    d->>s: navigate spirekey.kadena.io/login?returnUrl=dapp.com/auth
+    u->>s: grant authorization `base-64-user`
+    s->>d: dapp.com/auth?user=base-64-user
+    d->>u: logged in with `base-64-user`
+```
+
+### Sign Transaction from dApp
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:94]:
```suggestion
### Sign a transaction from the application

The following diagram illustrates the process for signing a transaction using a Kadena SpireKey account.
```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+### Kadena SpireKey authenticate/login
+
+#### Authentication request
+
+A GET request needs to be send to Kadena SpireKey
+
+```
+spirekey.kadena.io/login?returnUrl=your-dapp.com/auth
+```
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:119]:
```suggestion

## Authentication request and response

To authenticate user requests for access to your application, your application must send a GET request to the Kadena SpireKey endpoint and receive a response.

### Authentication request

Send a GET request to the Kadena SpireKey `login` endpoint using the following format:

https://spirekey.kadena.io/login?returnUrl=your-dapp.com/auth

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+- host `spirekey.kadena.io`
+- path `/login`
+- querystring parameters
+  - `returnUrl` the url where the user will be redirected to after
+    authentication.
+
+#### Authentication Response
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:128]:
```suggestion
### Authentication response

Your application receives a response with the following payload:

```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+  - `pendingTxIds`: `Array<string>` - List of transaction Request Keys. One of:
+    - Account Creation Transaction id: the account is being minted, and the dapp
+      can use this to verify when this is completed.  
+      The `user` object, can already be used to prepare transactions for the
+      user.
+
+### Kadena SpireKey Sign
+
+#### Sign Request
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:152]:
```suggestion
## Sign request and response

To sign transactions for your application, your application must send the transaction to the Kadena SpireKey endpoint and receive a response.

### Sign request

Send a request to the Kadena SpireKey `sign` endpoint using the following format:

https://spirekey.kadena.io/sign?transaction=tx-base64&returnUrl=your-dapp.com/auth
```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+```
+spirekey.kadena.io/sign?transaction=tx-base64&returnUrl=your-dapp.com/auth
+```
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:155]:
```suggestion
```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+- path `/sign`
+- querystring parameters
+  - `transaction`: [`IUnsignedCommand`]() a base64 encoded transaction
+  - `returnUrl` the url where the user will be redirected to after signing
+  - `optimistic`: `boolean` - when `true`, Kadena SpireKey will return a signed
+    transaction, even when there are pending transactions.  
+    The response will include `pendingTxIds`.
+
+#### Sign Response
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:167]:
```suggestion
### Sign response

Your application receives a response with the following payload:

```

**Review by lsgunnlsgunn — COMMENTED** [review]
Some suggestions but still pretty rough.

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
+---
+title: Kadena SpireKey User Guide
```
[the remark this answers is not in the document]
Comment (alber70g) [reply packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:2]: I changed it to Authentication and Signing

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+```
+
+## Specification
+
+### OAuth authenticate
+
+#### Authentication request
+
+A GET request needs to be send to Kadena SpireKey
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:189]:
```suggestion
## Authentication request and response

To authenticate users for your application, your application must send a GET request to the Kadena SpireKey `login` endpoint and receive a response.

### Authentication request

Send a GET request to Kadena SpireKey using the HTTPS protocol.

```

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+    w->>s: `WebAuthn-token`
+    s->>k: create account with `WebAuthn-token`
+  alt Validation by blockchain
+    k->>k: executing transaction, mining block
+    k->>s: account created `c:my-account`
+  else Validation failed
+    k->>s: transaction invalid
+  end
+```
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:144]: Need some descriptive text to explain what the diagram is trying to convey.

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 16]
```diff
@@ -0,0 +1,261 @@
...
+  participant d as dApp
+    u->>d: Login with SpireKey
+  create participant s as Kadena SpireKey
+    d->>s: navigate https://spirekey.kadena.io/login?returnUrl=https://your-dapp.com/auth
+    s-->>u: request authorization
+    u->>s: grant authorization `base-64-user`
+    s->>d: https://your-dapp.com/auth?user=base-64-user
+    d->>u: logged in with `base-64-user`
+```
+
```
Comment (lsgunnlsgunn) [inline packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:162]: Need some descriptive text to explain what the diagram is trying to convey.

**Review by lsgunnlsgunn — COMMENTED** [review]
Still very preliminary doc

File: packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md [hunk 17]
```diff
@@ -0,0 +1,222 @@
...
+    u->>d: buy cookie
+    d->>d: create buy-cookie-transaction `tx-base64`
+  create participant s as Kadena SpireKey
+    d->>s: spirekey.kadena.io/sign?transaction=tx-base64&returnUrl=dapp.com/auth
+    u->>s: grant signature `tx-base64-signature`
+    s->>d: dapp.com/auth?transaction=tx-base64-signature
+  create participant k as Kadena Blockchain
+    d->>k: send transaction
+```
+
```
[the remark this answers is not in the document]
Comment (alber70g) [reply packages/apps/docs/src/docs/kadena/wallets/kadena-spirekey/index.md:181]: I'm writing a minimal implementation as a showcase. Maybe we can take parts from there: https://github.com/kadena-community/kadena.js/pull/1768/files#diff-899d8c6b610edd0a215fe8b8190afe883bfea0c1173a5fe135540041fc05b875

File: packages/apps/docs/src/docs/kadena/wallets/index.md [hunk 18]
```diff
@@ -18,6 +18,26 @@ started.
 
 ## Software Wallets
 
+### Kadena SpireKey (WebAuthN)
+
+The official implementation of the draft
+[KIP-00XX SpireConnect](https://github.com/kadena-community/webauthn-wallet/blob/main/KIP-00XX.md)
```
Comment (Randynamic) [inline packages/apps/docs/src/docs/kadena/wallets/index.md:24]:
```suggestion
[KIP-0030 Kadena SpireConnect](https://github.com/kadena-community/webauthn-wallet/blob/main/KIP-0030.md)
```

**Review by Randynamic — APPROVED** [review]
